dropshot/src/handler.rs

```diff
@@ -1414,6 +1503,33 @@ impl From<HttpResponseUpdatedNoContent> for HttpHandlerResult {
     }
 }
 
+/**
+ * `HttpResponseUpgraded<T: Serialize>` wraps an object of any serializable type.
+ * It denotes an HTTP 101 "Switching Protocols". This is used for the response from
+ * connections upgraded to a WebSocket.
+ */
+pub struct HttpResponseUpgraded(pub headers::SecWebsocketKey);
```

Comment:
Should we make the name indicate it's websocket specific in case we later do want to allow arbitrary upgrades?

Reply:
can do!